students/YingGuo/session09/cli_main.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+
+from donor_models import *
+
+if __name__ == "__main__":
+    dc = DonorCollection()
+    while True:
+        try:
+            user_input = int(input("\
+            1. add new donor\n\
+            2. add donation to existing donor\n\
+            3. full report\n\
+            4. send thank you\n\
+            5. quit"))
+
+            if user_input == 1:
+                input_name = input("What's the name of new donor?")
+                dc.add_new_donor(input_name)
+                print(dc.__init__())
```

Comment:
Hmm... Why are you doing this? `dc.__init__()` doesn't return anything, so nothing (except `None`) will be printed. Besides, calling `dc.__init__()` will reset the underlying `dc.__donors` dictionary that stores the added donors. Is that really what you wanted? You should have tested more involved workflow, not just individual features separately.